src/main/java/org/usfirst/frc/team4099/robot/subsystems/Arm.kt

```diff
@@ -0,0 +1,208 @@
+         */
+        override fun onLoop() {
+            synchronized(this@Arm) {
+                // if quadrature velocity is zero, set to hold
+                // after 0.5 seconds of hold, actuate the brake and prevent movement
+                // 0.5 seconds after actuating the brake, set to stationary
+                // once the user tries to move, un-actuate the brake
+                if (masterSRX.sensorCollection.quadratureVelocity == 0) {
+                    movementState = Arm.MovementState.HOLD
+                    armState = Arm.ArmState.STILL
+                }
+                if (movementState == Arm.MovementState.HOLD && Timer.getFPGATimestamp() - holdTime >= 0.5) {
+                    brake.set(DoubleSolenoid.Value.kReverse)
+
+                }
+                if (movementState != Arm.MovementState.STATIONARY && brake.get() == DoubleSolenoid.Value.kReverse) {
+                    brakeTime = Timer.getFPGATimestamp()
+                    brake.set(DoubleSolenoid.Value.kForward)
+                }
+                if (Timer.getFPGATimestamp() - brakeTime < 0.5) {
+                    return
+                }
+                //if current time - init time > 0.5 : skip loop: set to stationary
+                if (movementState == Arm.MovementState.STATIONARY) {
+                    brake.set(DoubleSolenoid.Value.kReverse)
+                } else if (movementState == Arm.MovementState.HOLD) {
+                    masterSRX.set(ControlMode.MotionMagic, ArmConversion.pulsesToRadians(masterSRX.sensorCollection.pulseWidthPosition))
+                } else {
+                    brake.set(DoubleSolenoid.Value.kForward)
+                    when (armState) {
+                        Arm.ArmState.LOW -> {
+                            setArmPosition(ArmState.LOW.targetPos)
+                        }
+                        Arm.ArmState.EXCHANGE -> {
+                            setArmPosition(ArmState.EXCHANGE.targetPos)
+                        }
+                        Arm.ArmState.HIGH -> {
+                            setArmPosition(ArmState.HIGH.targetPos)
+                        }
+                            //masterSRX.sensorCollection.quadratureVelocity
+                        Arm.ArmState.VELOCITY_CONTROL -> { }
+                        Arm.ArmState.STILL -> {
+                            masterSRX.set(ControlMode.MotionMagic, ArmConversion.pulsesToRadians(masterSRX.sensorCollection.pulseWidthPosition))
+                        }
+                    }
+                }
+
+                if (masterSRX.sensorCollection.quadratureVelocity > 0) {
+                    movementState = Arm.MovementState.UP
+                } else if (masterSRX.sensorCollection.quadratureVelocity == 0) {
+                    //movementState = Arm.MovementState.HOLD
+                    if (movementState != Arm.MovementState.STATIONARY && brake.get() == DoubleSolenoid.Value.kReverse) {
```

Comment:
This check would fail whenever the bot is in `STATIONARY` and would instantly shift into `HOLD`.